### PR TITLE
Use AVFrame info for height and width in GPU APIs

### PR DESCRIPTION
**Pull request by NicolasHug**
instead of relying on metadata, like what we previously did.

Addresses https://github.com/pytorch/torchcodec/pull/339#discussion_r1832919983

### CODE REVIEW

File: src/torchcodec/decoders/_core/CudaDevice.cpp [hunk 1]
```diff
@@ -222,11 +219,10 @@ void convertAVFrameToDecodedOutputOnCuda(
   // Use the user-requested GPU for running the NPP kernel.
   c10::cuda::CUDAGuard deviceGuard(device);
 
-  auto start = std::chrono::high_resolution_clock::now();
+  NppiSize oSizeROI = {width, height};
+  Npp8u* input[2] = {src->data[0], src->data[1]};
```
Comment (NicolasHug) [inline src/torchcodec/decoders/_core/CudaDevice.cpp:223]: Just a drive-by, I moved this down closer to where they're used

**Review by ahmadsharif1 — APPROVED** [review]
